lib/patches/mailer_patch.rb

```diff
@@ -0,0 +1,79 @@
+require_dependency 'mailer'
+
+module Patches
+  module MailerPatch
+    def self.included(base)
+      base.send(:include, InstanceMethods)
+      base.class_eval do
+        alias_method :super_issue_edit, :issue_edit
+        alias_method :issue_edit, :patched_issue_edit
+
+        alias_method :super_message_posted, :message_posted
+        alias_method :message_posted, :patched_message_posted
+      end
+    end
+
+    module InstanceMethods
+      def patched_issue_edit(user, journal)
+        issue = journal.journalized
+
+        redmine_headers 'Project' => issue.project.identifier,
+                        'Issue-Tracker' => issue.tracker.name,
+                        'Issue-Id' => issue.id,
+                        'Issue-Author' => issue.author.login,
+                        'Issue-Assignee' => assignee_for_header(issue)
+        redmine_headers 'Issue-Priority' => issue.priority.name if issue.priority
+        message_id journal
+        references issue
+
+        @author = journal.user
```

Comment:
I see that the function `issue_edit` and `message_posted` are heavily different from original. Any reason why?

Reply:
I made the changes to `issue_edit` and `message_posted` because I needed to introduce the functionality that controls whether the email sends the teaser or the whole note. My goal was to keep the changes as minimal as possible while ensuring the email content behaves as expected. In my last commit I also refactored the logic for checking the value of the dropdown and computing the email content by moving it into a separate helper function (`mail_content_with_teaser`) to make the code cleaner and easier to maintain. That said, if you think there's a better way to handle this or if the changes feel too extensive, I'm open to revisiting the code further. Let me know if you have any specific ideas or suggestions on how you'd like it adjusted.